### ISSUE TITLE
Accessory will not add to HomeKit on iOS 18.2

### ISSUE THREAD

**Issue by ThomasKWood** [grzegorz914/homebridge-denon-tv#304]
I see this issue has been made multiple times now but they have all gone stale. I'm hoping that opening a new one will get this properly addressed.

I've followed the Wiki instructions on adding the Denon Receiver to HomeKit and I have had success in the past before updating to iOS 18.x.

However, now that I am on 18.2 the functionality of adding the receiver manually by selecting it from the More options list seems to no longer work.

I've tried the following:
adding with bridge mode off, using a config that I know worked before, restarting the homebridge host, and restarting the Receiver.

Config:
{
    "devices": [
        {
            "name": "DENON",
            "host": "10.0.0.2",
            "port": 80,
            "generation": 1,
            "zoneControl": 0,
            "getInputsFromDevice": true,
            "getFavoritesFromDevice": true,
            "getQuickSmartSelectFromDevice": true,
            "inputsDisplayOrder": 0,
            "buttons": [
                {
                    "name": "PC",
                    "reference": "0SIGAME",
                    "displayType": 2
                }
            ],
            "buttonsZ2": [
                {
                    "displayType": 0
                }
            ],
            "buttonsZ3": [
                {
                    "displayType": 0
                }
            ],
            "sensorPower": false,
            "sensorVolume": false,
            "sensorMute": false,
            "sensorInput": false,
            "sensorSurrounds": [
                {
                    "displayType": 0
                }
            ],
            "infoButtonCommand": "MNMEN ON",
            "masterPower": false,
            "masterMute": false,
            "masterVolume": false,
            "volumeControlNamePrefix": false,
            "volumeControl": 0,
            "volumeMax": 100,
            "refreshInterval": 5,
            "enableDebugMode": true,
            "disableLogInfo": false,
            "disableLogDeviceInfo": false,
            "disableLogConnectError": false,
            "restFul": {
                "enable": false,
                "debug": false
            },
            "mqtt": {
                "enable": false,
                "debug": false,
                "auth": false
            }
        }
    ],
    "_bridge": {
        "username": "****************",
        "port": 32947
    },
    "platform": "DenonTv"
}

Logs:
[12/31/2024, 10:45:22 AM] [homebridge-denon-tv] Launched child bridge with PID 8843
[12/31/2024, 10:45:23 AM] [homebridge-denon-tv] Loaded homebridge-denon-tv v5.2.3 child bridge successfully
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Device: 10.0.0.2 DENON S720W, Connect Success.
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] -------- DENON S720W --------
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Manufacturer: Denon
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Model: *AVR-S720W
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Zones: 2
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Control: Main Zone
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Firmware: 0
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Api version: 300
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Serialnr: *********************
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] ----------------------------------
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Device: 10.0.0.2 DENON S720W, Published as external accessory.
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Device: 10.0.0.2 DENON S720W, Impulse generator started.
[12/31/2024, 10:45:24 AM] DENON S720W A87F is running on port 37389.
[12/31/2024, 10:45:24 AM] Please add [DENON S720W A87F] manually in Home app. Setup Code: 144-44-013
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Device: 10.0.0.2 DENON S720W, Power: ON.
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Device: 10.0.0.2 DENON S720W, Input Name: DVD.
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Device: 10.0.0.2 DENON S720W, Reference: DVD.
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Device: 10.0.0.2 DENON S720W, Volume: -50dB.
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Device: 10.0.0.2 DENON S720W, Mute: OFF.
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Device: 10.0.0.2 DENON S720W, Volume Control Type: Absolute.
[12/31/2024, 10:45:24 AM] [homebridge-denon-tv] Device: 10.0.0.2 DENON S720W, Picture Mode: OFF.

[12/31/2024, 12:24:38 PM] [homebridge-denon-tv] Device: 10.0.0.2 DENON, debug: State: {
  "Power": {
    "value": "OFF"
  },
  "InputFuncSelect": {
    "value": "DVD"
  },
  "VolumeDisplay": {
    "value": "Absolute"
  },
  "MasterVolume": {
    "value": -48.5
  },
  "Mute": {
    "value": "off"
  }
}.

**Comment by grzegorz914** [comment]
Try troubleshooting section in wiki

**Comment by ThomasKWood** [comment]
Yep I did the config UI method, restarted, and then reinstalled. Same issue persists

**Comment by grzegorz914** [comment]
May be try to change device name in config, as you see in the log all is published correct.

**Comment by ThomasKWood** [comment]
I tried naming the accessory Denon and Denon S720W in the config with no luck. I agree, everything looks to be working on the Homebridge side but it will not add to HomeKit

**Comment by grzegorz914** [comment]
Also try to reboot your network router, switch, wlan, etc.

**Comment by ThomasKWood** [comment]
Other plugins are communicating and adding to HomeKit just fine so I don't see how restarting the network infrastructure will fix it.

**Comment by grzegorz914** [comment]
Very strange, just removed device on my side and add it again without any problem.
![IMG_1050](https://github.com/user-attachments/assets/94b359bd-8108-4961-99ab-3d1856c9e0d4)

**Comment by ThomasKWood** [comment]
Do you have Advanced Data Protection or Private Relay On? I am wondering if those may be interfering.

**Comment by grzegorz914** [comment]
I can not reprodukcje this issue, on my side all is correct.

**Comment by ThomasKWood** [comment]
After rebuilding my Homebridge several times and trying out different older versions of this plug-in, I simply gave up. I followed all the steps correctly from setting up Homebridge itself to setting up the plug-in and even tried disabling Advanced Data Protection and Private Relay, but none of it helped. From the issue reports, it’s clear that I’m not the only one experiencing this problems. What might be working for you doesn't seem to be the case on new versions of iOS and iPadOS for some users.

Ultimately, I decided to use Home Assistant, with its built-in support for Denon receivers, to meet my needs.
